Backends/CLX/input.lisp

```diff
@@ -223,9 +223,14 @@
                              :sheet sheet
                              :x x :y y :width width :height height))))
       (:map-notify
-       (make-instance 'window-map-event :sheet sheet))
+       (if (and (typep sheet 'top-level-sheet-pane)
+                (eq (frame-state (pane-frame sheet)) :shrunk))
+           (make-instance 'window-manager-deiconify-event :sheet sheet)
+           (make-instance 'window-map-event :sheet sheet)))
       (:unmap-notify
-       (make-instance 'window-unmap-event :sheet sheet))
+       (if (= 3 (car (xlib:get-property window :WM_STATE)))
```

Comment:
perhaps we should define appropriate constants (i.e in clx) for states - given they are standard across different x11 implementations - of course if they are not already defined, if they are then we just use them.

Reply:
I didn't find the constant in clx so I defined them in clim-clx